private/bufpkg/bufprotoplugin/response_writer.go

```diff
@@ -110,48 +111,43 @@ func writeInsertionPoint(
 	// bufio.Scanner.Scan() inline
 	newline := []byte{'\n'}
 	var found bool
-	for targetScanner.Scan() {
+	for i := 0; targetScanner.Scan(); i++ {
+		if i > 0 {
+			// These writes cannot fail, they will panic if they cannot allocate.
+			_, _ = postInsertionContent.Write(newline)
```

Comment:
Create a newline only if we are going to append more content to the file. This removes the need to trim after the buffer is finalized.